flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/RowTypeTest.scala

```diff
@@ -129,4 +128,17 @@ class RowTypeTest extends RowTypeTestBase {
       "null"
     )
   }
+
+  @Test
+  def testUnsupportedCast(): Unit = {
+    expectedException.expect(classOf[ValidationException])
+    expectedException.expectMessage(
+      "Unsupported cast from Row(f0: String, f1: Boolean) to Long")
+
+    testTableApi(
+      'f5.cast(DataTypes.BIGINT()),
```

Comment:
This test case can't reproduce the problem which reported by this issue. You should add a test case which cast a ROW to another ROW, e.g. 

```scala
testSqlApi("CAST(f5 AS ROW<f0 INT, f1 STRING>)", "")
```

Reply:
Ah... yes, this is table api, we want to fix sql api.

Reply:
You should keep table api and sql api both.

Reply:
OK, Added a sql api test.